### PR TITLE
Fix SKOptLearner for multi variate domain (issue #233)

### PR DESCRIPTION
**Pull request by caenrigen**
This fixes issue #233 

Tests checked. Per-commit checked.
Implemented additional test for this case.

Changed `data` type to `OrderedDict` to comply with other learners.

### CODE REVIEW

**Comment by caenrigen** [comment]
Strange that the `linting_test` failed
I get not error when running locally `pre-commit run --all-files`

How can it be fixed?

**Comment by jbweston** [comment]
> Strange that the `linting_test` failed
> I get no error when running locally `pre-commit run --all-files`
> 
> How can it be fixed?

When I click through and look at the test failure I see that it's because `black` reformatted some files.
I am not sure why `pre-commit` succeeds on your machine. When I check out your branch locally and run `pre-commit run --all-files` the `black` check indeed fails.

**Comment by caenrigen** [comment]
> > Strange that the `linting_test` failed
> > I get no error when running locally `pre-commit run --all-files`
> > How can it be fixed?
> 
> When I click through and look at the test failure I see that it's because `black` reformatted some files.
> 
> I am not sure why `pre-commit` succeeds on your machine. When I check out your branch locally and run `pre-commit run --all-files` the `black` check indeed fails.

You are right there was some message about black the first time I run pre-commit, I didn't notice it, I see the formatting changed in the file and it doesn't complain anymore.
Not sure how things work, if I do a new commit and push to the same branch you will be able to merge it?

File: adaptive/learner/skopt_learner.py [hunk 1]
```diff
@@ -26,18 +28,23 @@ class SKOptLearner(Optimizer, BaseLearner):
     def __init__(self, function, **kwargs):
         self.function = function
         self.pending_points = set()
-        self.data = {}
+        self.data = collections.OrderedDict()
```
Comment (basnijholt) [inline adaptive/learner/skopt_learner.py]: In Python >= 3.6 all dicts are ordered in principle, but it is good to make it explicit.

**Comment by basnijholt** [comment]
@jbweston, should we just merge this into master? Seems like we didn't put anything on https://github.com/python-adaptive/adaptive/tree/stable-0.9.

**Comment by caenrigen** [comment]
> LGTM! Thanks a lot for your contribution!

Thank you, guys, for the nice package! 😃

**Comment by jbweston** [comment]
> @jbweston, should we just merge this into master? Seems like we didn't put anything on https://github.com/python-adaptive/adaptive/tree/stable-0.9.

Depends; do we want to release v0.9.1? So far we have not been properly adhering to semver, which I think is fine in our pre-1.0 no-backwards-compatibility state. I think we can elide keeping `stable-*` branches around always put things into the next minor release.

**Comment by basnijholt** [comment]
I think we can skip a 0.9.1. So let's merge!

**Pull request merged**